sdk/src/main/java/io/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -216,8 +218,14 @@ public TransactionSerializer serializer() {
         return EthereumTransactionSerializer.getSerializer();
     }
 
+    @Deprecated
     @Override
     public void semanticValidity() throws TransactionSemanticValidityException {
+        semanticValidity(0);
```

Comment:
possible problem: the method semanticValidity() (without the consensus epoch parameter) is still used in SidechainBlockBase, which is called by the validators before appending the block to the history.
With this change we could have a block that pass the history validation but then is marked as invalid in the state apply (where is called semantiValidiry(consensusEpoch))

Reply:
You're right. I have moved the transaction validation inside a method in SidechainBlockBase that is overridden by AccountBlock, so the new version of semanticValidity can be called. The old version of semanticValidity now throws an exception , so it cannot be called anymore in account model.